tests/test_file_utils.py

```diff
@@ -345,29 +345,49 @@ class TestGetEndian:
         "content, big_endian_magic, expected",
         [
             pytest.param(
-                b"\xff\x00\x00\x10", 0x100000FF, Endian.LITTLE, id="valid_little_endian"
+                b"\xff\x00\x00\x10",
+                0x100000FF,
+                Endian.LITTLE,
+                id="valid_little_endian_4",
             ),
             pytest.param(
-                b"\x10\x00\x00\xff", 0x100000FF, Endian.BIG, id="valid_big_endian"
+                b"\x10\x00\x00\xff", 0x100000FF, Endian.BIG, id="valid_big_endian_4"
             ),
+            pytest.param(
+                b"\xff\x00", 0x00FF, Endian.LITTLE, id="valid_little_endian_2"
+            ),
+            pytest.param(b"\x10\x00", 0x1000, Endian.BIG, id="valid_big_endian_2"),
         ],
     )
     def test_get_endian(self, content: bytes, big_endian_magic: int, expected: Endian):
         file = File.from_bytes(content)
-        assert get_endian(file, big_endian_magic) == expected
+        assert get_endian(file, big_endian_magic, endian_len=len(content)) == expected
```

Comment:
This is failing the type check, as it is not known to the type checker if `len(content)` is surely one of `2` and `4`.

This test needs to be split into 2, to test, one of magic length 2, and for magic length 4.

(Which also shows that `get_endian()` should have a new version instead of being extended, keeping the original and defining a new one for 2 byte magic)

Reply:
What about leaving the original `get_endian()` as is, and introducing a new `get_endian_short()`?

I prefer 2 smaller function, than one bigger, which is also harder to use/understand.

Reply:
implemented `get_endian_short` and adapted code and tests, should be easier to navigate now. Commit message has been adjusted too.